### ISSUE TITLE
graph node process will  pending when there are many transactions in block.

### ISSUE THREAD

**Issue by huangyisan** [graphprotocol/graph-node#3106]
**Do you want to request a *feature* or report a *bug*?**
Neither.

**What is the current behavior?**

Our graph-node was deploy on aws EC2 with m5.xlarge type. And the postgresql is db.t4g.xlarge.

The mainnet is bsc. And we deploy our own bsc full node.

When a large number of transactions persist in blocks(shows in https://bscscan.com/blocks?p=1), we observe the logs and find that the logs out put get stuck, like all the component are stop, and then the value of blocks_behind  goes up.

At that time, our business will be affected for a short period of time.

Also we check our bsc node, it can get the nearest block normally.

Here is our graph node env configurations: 

export GRAPH_ETHEREUM_TARGET_TRIGGERS_PER_BLOCK_RANGE=500
export ETHEREUM_BLOCK_BATCH_SIZE=50

export GRAPH_MAX_IPFS_MAP_FILE_SIZE=1024000000
export GRAPH_MAX_IPFS_CACHE_SIZE=200
export GRAPH_ENTITY_CACHE_SIZE=50000

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**

We want to know the cause of the stuck and fix it.

Thanks.

**Comment by maoueh** [comment]
Would you be able to provide the logs in question so we can see what is happing? Specially if you are able to reproduce using DEBUG or TRACE logs (trace logs might be too verbose if you need to wait a long time before it reproduces).

**Comment by huangyisan** [comment]
Thanks maoueh!  we will provide the logs as soon as possible.

**Comment by huangyisan** [comment]
Hi, I have post the graph_node.log, the log level is INFO. Please see the attachement.

1. In the graph_node.log ,we can see from "Jan 02 14:00:18.984" to "Jan 02 14:10:45.616" the value of "blocks_behind" is increasing, And then it starts to decreasing.

2. BlockIngestor component does not have any log output at 14:04, 14:07, 14:09, 14:11. I think maybe SubgraphInstanceManager is too busy, or other reasons. But cpu utilization is low from aws monitor.

At the same time ,I have checked the transactions on "https://bscscan.com/blocks", from block 14015207 (2022-01-02 13:59:32 UTC) to 14015309 (2022-01-02 14:06:22 UTC), the number of transactions per block is significantly higher compared to the general case, with some even exceeding 1000.

Now I've prepared a machine with the log level set to debug, ready to capture the logs of the next stuck.

[graph_node.log](https://github.com/graphprotocol/graph-node/files/7799359/graph_node.log)

**Comment by maoueh** [comment]
From what I'm reading from the code around the logs you uploaded, it could be that the RPC now is long to send you back information about the blocks because indeed, there is a hell lot of transactions in those.

There have been some improvements recently on the `master` branch where transactions are retrieved concurrently based on the value of `MAX_CONCURRENT_JSON_RPC_CALLS`. That has been done with BSC in mind where the amount of transactions to fetch is really high. 

I will wait for more logs from you but for now, my current hypothesis would be that it takes a long time to ingest the blocks from the JSON-RPC which causes some delay in processing.

**Comment by huangyisan** [comment]
Thanks maouth!  I have caught some new logs in DEBUG level. 

from block 14076874 (2022-01-04 17:29:52 UTC) to 14077016 (2022-01-04 17:36:58 UTC) also have a lot of transactions in per block .

In the "graphnode_debug.log",  from  ”Jan  4 17:35:13“ to "Jan  4 17:38:43" only record "no chain head update for 30 seconds, polling for update,sgd: 3".

[graphnode_debug.log](https://github.com/graphprotocol/graph-node/files/7809640/graphnode_debug.log)

**Comment by azf20** [comment]
yes @huangyisan I think this is related to the batch RPC calls, as @maoueh says there is an optimisation on the latest `master` branch that address this, you can try it using the `GRAPH_EXPERIMENTAL_FETCH_TXN_RECEIPTS_CONCURRENTLY` feature flag (set to any value)

**Comment by huangyisan** [comment]
Thanks all, I have set the MAX_CONCURRENT_JSON_RPC_CALLS to 50, and GRAPH_EXPERIMENTAL_FETCH_TXN_RECEIPTS_CONCURRENTLY to 50. Hope it works :)
